mmn/mmn.go

```diff
@@ -0,0 +1,77 @@
+package mmn
+
+import (
+	"github.com/mysteriumnetwork/node/core/connection"
+	"github.com/mysteriumnetwork/node/core/service"
+	"github.com/mysteriumnetwork/node/eventbus"
+	"github.com/mysteriumnetwork/node/identity"
+
+	"github.com/rs/zerolog/log"
+)
+
+type MMN struct {
+	collector *Collector
+	client    *client
+	eventBus  eventbus.EventBus
+}
+
+func NewMMN(collector *Collector, client *client, eventBus eventbus.EventBus) *MMN {
+	return &MMN{collector, client, eventBus}
+}
+
+func (m *MMN) SubscribeToEvents() {
+	m.subscribeToRegistrationEvent()
+	m.subscribeToProviderEvent()
+	m.subscribeToConsumerEvent()
+}
+
+func (m *MMN) Register(identity string) error {
+	m.collector.SetIdentity(identity)
+
+	return m.client.RegisterNode(m.collector.GetCollectedInformation())
+}
+
+func (m *MMN) UpdateNodeType(isProvider bool) error {
```

Comment:
might want to put this under lock as this will have tendency to race since you're subscribing in async fashion